Integrations/integration-ProtectWise.yml

```diff
@@ -201,101 +187,300 @@ script:
-                Method: 'GET'
-            },
-            params.insecure
-            );
-        if (res.StatusCode !== 200) {
-            throw 'Failed to execute request get sensors: ' + res.StatusCode + ', body: ' + res.Body;
-        }
-        return res.Body;
-    };
+
     var parseTime = function(time) {
         if ((typeof time === 'string' || time instanceof String) && (time.indexOf("-") >= 0 || time.indexOf("/") >= 0 )){
             var d = new Date(time);
             return d.getTime();
         }
         return time;
-    }
+    };
+
+
+    var createIncidentFromEvent = function(event) {
+        var keys = Object.keys(event);
+        var labels = [];
+        for (var i = 0; i<keys.length; i++) {
+            val = event[keys[i]];
+            if (TIME_FIELDS.indexOf(keys[i]) > -1) {
+                val = convertTimestampToString(val);
+            }
+            labels.push({'type': keys[i], 'value': String(val)});
+        }
+        return {
+            "name": event.message,
+            "labels": labels,
+        };
+    };
+
+    var token = getToken();
     switch (command) {
         case 'test-module':
-            var token = getToken();
            if (token && token.length > 0){
                 return true;
             }
             return false;
+        case 'fetch-incidents':
+            var lastRun = getLastRun();
+            var startTime = (lastRun && lastRun.startTime ? lastRun.startTime : (new Date().getTime()- 1000 * 60 ));
+            var data = eventsSearch(startTime , new Date().getTime() , params.eventType, params.killChainStage, params.threatLevel,
+                                    params.threatCategory, undefined,undefined,undefined,undefined,undefined,undefined,undefined,token);
+            var res = JSON.parse(data).events;
+            var incidents = [];
+            for (var i = 0; i < res.length; i++) {
+                var skip = true;
+                if (params.messageFilter && res[i].message) {
+                    filters = params.messageFilter.split(',');
+                    for (var j = 0; j < filters.length; j++) {
+                        if (filters[j] && filters[j].length > 0 &&  res[i].message.toLowerCase().indexOf(filters[j].toLowerCase()) > -1) {
+                            skip = false;
+                            break;
+                        }
+                    }
+                } else {
+                    skip = false;
+                }
+                if(!skip) {
+                    incidents.push(createIncidentFromEvent(res[i]));
+                }
+            }
+            setLastRun({startTime: startTime});
+            return JSON.stringify(incidents);
         case 'sensors':
-            var token = getToken();
-            return {Type: 1, Contents: sensors(args.sensorId,token), ContentsFormat: 'json'};
+        case 'protectwise-show-sensors':
+            var url = params.url+'sensors';
```

Comment:
In most integrations we protect ourselves from cases placed / didnt place the `/`